sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -122,24 +129,24 @@ object AnalysisContext {
  */
 class Analyzer(
     override val catalogManager: CatalogManager,
+    v1SessionCatalog: SessionCatalog,
```

Comment:
One downside of this approach is that a `SessionCatalog` instance which is different from the one stored in `CatalogManager`. Since `CatlalogManager` updates the current database of the session catalog, it can be out of sync. Please let me know if this approach is fine.

Reply:
hmm, what's the upside of doing this?

Reply:
`v1SessionCatalog` is now private in `CatalogManager`. Since `Analyzer` uses `v1SessionCatalog` we need to pass this separately.

Reply:
We don't need to make `v1SessionCatalog` private in `CatalogManager`. We only need to make `sessionCatalog` and `defaultCatalog` private. cc @rdblue